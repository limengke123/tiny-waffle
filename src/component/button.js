import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

// class Button extends React.PureComponent {
//     static propTypes = {
//         text: PropTypes.string,
//         onClick: PropTypes.func
//     }
//
//     static defaultProps = {
//         text: '',
//         onClick() {}
//     }
//
//     render() {
//         console.log('count button render')
//         const { text, onClick: onClickProps } = this.props
//         return (
//             <button type="button" onClick={onClickProps}>
//                 {text}
//             </button>
//         )
//     }
// }

function Button({ text, onClick }) {
    console.log('count button render')
    return (
        <button type="button" onClick={onClick}>
            {text}
        </button>
    )
}

const MemoButton = React.memo(Button)

export { MemoButton, Button }
