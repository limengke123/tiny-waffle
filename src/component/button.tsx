import React from 'react'

export interface ButtonProps {
    text: string
    onClick: () => void
}

export const Button: React.FunctionComponent<ButtonProps> = function({
    text,
    onClick
}) {
    return (
        <button type="button" onClick={onClick}>
            {text}
        </button>
    )
}

const MemoButton = React.memo(Button)

export { MemoButton }
