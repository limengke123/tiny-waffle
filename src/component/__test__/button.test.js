import React from 'react'
import { render, mount } from 'enzyme'
import { MemoButton } from '../button'

describe('Button', () => {
    it('renders correctly', () => {
        const wrapper = render(<MemoButton text="按钮文案" />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should handle click event correctly', () => {
        const onClick = jest.fn()
        const wrapper = mount(<MemoButton text="测试按钮" onClick={onClick} />)
        wrapper.simulate('click')
        expect(onClick).toHaveBeenCalled()
    })
})
