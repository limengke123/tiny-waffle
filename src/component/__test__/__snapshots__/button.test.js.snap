// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  type="button"
>
  按钮文案
</button>
`;
